core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -115,7 +116,11 @@ protected void doRefresh() {
   public void commit(TableMetadata base, TableMetadata metadata) {
     // if the metadata is already out of date, reject it
     if (base != current()) {
-      throw new CommitFailedException("Cannot commit: stale table metadata");
+      if (base != null) {
+        throw new CommitFailedException("Cannot commit: stale table metadata");
+      } else {
+        throw new AlreadyExistsException("Table already exists: %s", tableName());
```

Comment:
This exception message reads a little funny to me, but I think in context it will make sense.

For clarification, what's happening here is that we entered some sort of create table transaction, and then at the end of the transaction, the table was created by some other process. Is that correct?

A comment might help make that more clear but I can see how the stack trace will make sense in context.

Reply:
I added a comment to make this clear.